src/errors.rs

```diff
@@ -27,6 +27,8 @@ pub enum InternalError {
     InternalInvariantFailed,
     #[error("Storage does not contain the requested item")]
     StorageItemNotFound,
+    #[error("Broadcast Message does not have the correct tag")]
+    IncorrectBroadcastMessageTag,
```

Comment:
todo: We're trying to consolidate errors into a smaller set that defines what action the calling application should take, so I recommend removing this and using the `ProtocolError` variant instead -- that's an indication that a different participant tagged the message incorrectly and it's not a problem the calling application has to deal with.

I think the issue was written before #268 removed this error type; you can see more discussion on the goals of the `InternalError` type in the comments of #157.